app/experimenter/nimbus-ui/src/lib/visualization/utils.tsx

```diff
@@ -38,22 +38,41 @@ export const getControlBranchName = (analysis: AnalysisData) => {
   if (external_config?.reference_branch) {
     return external_config.reference_branch;
   }
-  const results = analysis.overall?.all || analysis.weekly?.all;
-  for (const [branchName, branchData] of Object.entries(results!)) {
-    if (branchData.is_control) {
-      return branchName;
+  const results = analysis.overall?.all || analysis.weekly?.all || {};
+  if (Object.keys(results).length > 0) {
+    for (const [branchName, branchData] of Object.entries(results)) {
+      if (branchData.is_control) {
+        return branchName;
+      }
+    }
+  }
+  // last option - try to find a unique branch name in the daily results
+  const daily = analysis.daily?.all || [];
+  if (daily.length > 0) {
+    const branches = new Set();
+    daily.forEach((point) => branches.add(point.branch));
```

Comment:
```suggestion
    const branches = new Set(daily.map((point) => point.branch));
```
Not a necessary change, but you could swap this to construct the set with an array of the branches on the daily points.